Targeting/Targeting.js

```diff
@@ -0,0 +1,54 @@
+class Targeting {
+  constructor() {
+    this.targets = [];
```

Comment:
In Lua there was two arrays because Lua does not have proper OOP, in JS there only needs to be one array `this.targets` and each targeting instance gets its own array